internal/cli/template_update.go

```diff
@@ -0,0 +1,233 @@
+package cli
+
+import (
+	"os"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateUpdateCommand struct {
+	*baseCommand
+
+	flagID   string
+	flagName string
+
+	flagSummary                    string
+	flagExpandedSummary            string
+	flagReadmeMarkdownTemplatePath string
+	flagWaypointHCLTemplatePath    string
+	flagTFCNoCodeModuleSource      string
+	flagTFCNoCodeModuleVersion     string
+	flagTags                       []string
+}
+
+func (c *ProjectTemplateUpdateCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+		return 1
+	}
+
+	var tref pb.Ref_ProjectTemplate
+	if name != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: name,
+		}
+	} else if c.flagID != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Id{
+			Id: c.flagID,
+		}
+		name = c.flagID
+	} else if c.flagName != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: c.flagName,
+		}
+		name = c.flagName
+	} else {
+		c.ui.Output("Missing project template name or id.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	checkResp, err := c.project.Client().GetProjectTemplate(ctx, &pb.GetProjectTemplateRequest{
+		ProjectTemplate: &pb.Ref_ProjectTemplate{
+			Ref: &pb.Ref_ProjectTemplate_Name{
+				Name: name,
+			},
+		},
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	if checkResp.ProjectTemplate == nil {
+		c.ui.Output(
+			"Project template %q does not exist", checkResp.ProjectTemplate.Name,
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	template := checkResp.ProjectTemplate
+
+	template.Name = name
+	if c.flagSummary != "" {
+		template.Summary = c.flagSummary
+	}
+	if c.flagExpandedSummary != "" {
+		template.ExpandedSummary = c.flagExpandedSummary
+	}
+
+	if c.flagReadmeMarkdownTemplatePath != "" {
+		rmt, err := os.ReadFile(c.flagReadmeMarkdownTemplatePath)
```

Comment:
In `Create` we check for `os.ErrNotExist` but we don't distinguish here or in the hcltemplate path check below

Reply:
Good catch, updated to match behavior with Create in `37e4c8269`